### DISCUSSION TITLE
Merge similar colors in Toonz level.

### DISCUSSION THREAD

**Discussion by F-Burning** [opentoonz/opentoonz#6796] [Roadmap]
I you paste a color from another level with a drawing, and repeat this along some frames, it creates new colors instead of paste it in one style. I think the skill to merge all similar levels in one, or paste drawings with a color into styles, would improve workflow.

**Comment by shun-iwasawa** [comment]
Currently when pasting frame or pasting raster selection from one toonz raster level to another, the used style in the pasted image will be merged to the destination level's existing style only when 1) the styles are identical (i.e. with the same names and colors) and 2) the style indices are the same.

**Comment by F-Burning** [comment]
> Currently when pasting frame or pasting raster selection from one toonz raster level to another, the used style in the pasted image will be merged to the destination level's existing style only when 1) the styles are identical (i.e. with the same names and colors) and 2) the style indices are the same.

True, interesting to know, thanks.

Anyway, sometimes you forget follow this steps because are very specific, and styles repeat on palette, so merge them would be a solution.

**Comment by F-Burning** [comment]
By the way, it's technically possible paste selections in different styles? I asked this time ago, but It not was clear for me if this is not possible to Opentoonz, or is a decision.

**Comment by RodneyBaker** [comment]
@F-Burning You may be interested in @Wolfinabowl's utility that can convert palettes, merge colors etc.
He has the uility (and others) on twitch.io and while it has a place for donations he's fine with people entering $0 to grab the program.  Off hand I don't recall the exact merging capabilities but it's a great utility to have standing by.  

Here's what he's written about the utility:

> Display, modify, create, and convert between a variety of palette formats.
> 
> Supported formats:
> 
> tpl - OpenToonz, Tahoma
> gpl - Krita, GIMP, MyPaint, and others
> pal - Aseprite and others
> col - Aseprite and others
> act - Aseprite and others
> hex - Aseprite and others
> cls (limited support) - Clip Studio Paint
> Create, modify, save, and load a variety of palette formats, allowing you to convert between them. Bring all your Krita/GIMP palettes into OpenToonz/Tahoma! (and vice versa)
> 
> Create new palettes from scratch by adding, removing ,altering and reordering entries.
> 
> Save palettes as png images to import into software such as Photoshop or Affinity Photo. Choose to display borders, border color, and the number of columns.
> 
> Extract palettes from images (Experimental). Works best with images that have a small number of colors and with no lossy compression. You can load images from local storage or from the web.
> 
> Generate new palettes by blending two colors, from random RGB ranges, and by sampling an image by clicking.
> 
> Note:
> cls support is currently limited and is planned to be improved/expanded in the future.

Note the blending of two colors line item.  This representes a stepping stone on the way to your requested feature.
Given support I imagine the Wolf that is in a bowl might be able to extend capabilities even farther.
The utility is created via Godot so there is no direct correlation between that code and Opentoonz but one never knows how the future might unfold.

There are certainly limitation in the utility.  
For instance, don't expect it to be able to manage mypaint and specialized brushes.

At any rate, here the link:  https://wolfinabowl.itch.io/palette-viewer-editor

**Comment by RodneyBaker** [comment]
> By the way, it's technically possible paste selections in different styles? I asked this time ago, but It not was clear for me if this is not possible to Opentoonz, or is a decision.

I don't know of any way to paste into a different style but selecting strokes and applying a new style is trivial; select the strokes and then the style.

Edit:  There is a way to 'paste into different styles' but hmmm... how to best explain it.
There are a bit too many nuances but you might wish to investigate keyframed color styles.
Those however will not likely work with special brushes which rather defeats the goal of 'pasting into other styles'.

**Comment by DarrenTAnims** [comment]
You can hold the control key and then drag the palette entries to a different order or to a different tab, @Bracket-H .

**Comment by F-Burning** [comment]
Thank you @RodneyBaker, looks useful. However (if i understood you right) my idea is not to make a mix (red + yellow = orange), but to merge 2 equal styles into one to have all the lines in a single color.

I was trying to change the de index style too yesterday @Bracket-H, but in Toonz raster there is no way.

I always liked how RETAS PRO HD works with different colors, because you can copy them and paste into another, or simply changes it with a click.

https://www.youtube.com/watch?v=cGTF2g5xmwg

I miss that versatility when drawing.

**Comment by RodneyBaker** [comment]
I've tagged this with roadmap primarily because there are some processes that merged similar colors but others that do not.  

A good next step forward will be to survey which processes require merging of similar color but currently do not.  

Also, while a defnite outlier, how can we allow for cases where the user does not want to merge similar colors when using these processes?

Adding details concerning what users are currently using as workarounds in production will be appreciated as that will help in better understanding the best way forward.

(NOTE:  This issue is ideal for Discussion so will likely be moved there)
